src/workspacer.FSharpConfig/FSharpCompiler.cs

```diff
@@ -0,0 +1,94 @@
+﻿namespace workspacer.FSharpConfig
+{
+    using FSharp.Compiler.CodeAnalysis;
+    using Microsoft.FSharp.Control;
+    using Microsoft.FSharp.Core;
+    using System;
+    using System.Collections.Generic;
+    using System.IO;
+    using System.Linq;
+    using System.Text.RegularExpressions;
+    using System.Threading;
+    using System.Reflection;
+
+    public static class FSharpCompiler
+    {
+        private static FSharpChecker checker =
+            FSharpChecker.Create(
+                FSharpOption<int>.None,
+                FSharpOption<bool>.None,
+                FSharpOption<bool>.None,
+                FSharpOption<LegacyReferenceResolver>.None,
+                FSharpOption<FSharpFunc<Tuple<string, DateTime>, FSharpOption<Tuple<object, IntPtr, int>>>>.None,
+                FSharpOption<bool>.None,
+                FSharpOption<bool>.None,
+                FSharpOption<bool>.None,
+                FSharpOption<bool>.None);
+
+        private static Regex requireRegex = new Regex("^#r @\"(.*)\"");
+
+        public static void CompileScript(string scriptFile, IList<string> references, string dllFile)
+        {
+            var compilerArgs = new List<string>
+            {
+                "fsc.exe",
+                "-a",
+                scriptFile,
+                "-o",
+                dllFile,
+                $"--lib:\"{Path.GetDirectoryName(Assembly.GetCallingAssembly().Location)}\"",
+            };
+
+            foreach (var r in references.Select(Path.GetDirectoryName).Distinct())
+            {
+                compilerArgs.Add($"--lib:\"{r}\"");
+            }
+
+            var result =
+                FSharpAsync.RunSynchronously(
+                    checker.Compile(compilerArgs.ToArray(), FSharpOption<string>.None),
+                    FSharpOption<int>.None,
+                    FSharpOption<CancellationToken>.None
+                );
```

Comment:
As far as I can see, this compiles and runs the config in one go.

Would it make more sense to run the compiler first, and *then* run the code in a subsequent call?

That might make it easier to handle and present meaningfull diagnostics-messages when the configuration either contains compilation-errors or has runtime-failures.